user/views.py

```diff
@@ -1,3 +1,86 @@
-from django.shortcuts import render
+import json, bcrypt
+import re
+
+from django.views import View
+from django.http import JsonResponse
+from .models import User, Gender, Grade, TermsAndCondition, Address
+from my_settings import SECRET_KEY
 
 # Create your views here.
+class SignupView(View): 
+    def validate_account(self, account):  
+        REGEX_ACCOUNT = '^[a-z0-9]{6,}$'
+
+        if not re.match(REGEX_ACCOUNT, account):
+            return False
+        return True
+
+
+    def validate_password(self, password):
+        REGEX_PASSWORD = '^[A-Za-z0-9@#$%^&+=]{10,}$'
+
+        if not re.match(REGEX_PASSWORD, password):
+            return False
+        return True
+
+    def validate_email(self, email): 
+        REGEX_EMAIL = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+
+        if not re.match(REGEX_EMAIL, email):
+            return False
+        return True
+    
+
+    def validate_phone_number(self, phone_number):
+        REGEX_PHONE_NUMBER = '^\d{3}?[-]\d{4}?[-]\d{4}$'
+        
+        if not re.match(REGEX_PHONE_NUMBER, phone_number):
+            return False
+        return True
+    
+
+    def post(self, request):
+        data = json.loads(request.body)
+
+        hashed_pw = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+
+        birth_date  = data.get('birth_date')
+        recommender = data.get('recommender')
+        event_name  = data.get('event_name')
+
+        try:
+
+            if not self.validate_account(data['account']):
+                return JsonResponse({"message":"INVALID_ID"}, status=400)
+
+            if User.objects.filter(account=data['account']).exists():
+                return JsonResponse({"message":"USER_EXIST"}, status=409)
+
+            if not self.validate_password(data['password']):
+                return JsonResponse({"message": "INVALID_PW"}, status=400)
+
+            if not self.validate_email(data['email']):
+                return JsonResponse({"message":"INVALID_EMAIL"}, status=400)
+
+            if not self.validate_phone_number(data['phone_number']):
+                return JsonResponse({"message":"INVALID_PHONE_NUMBER"}, status=400)
+
+            else:
+                User.objects.create(
+                    account             = data['account'],
+                    password            = hashed_pw,
+                    name                = data['name'],
+                    email               = data['email'],
+                    phone_number        = data['phone_number'],
+                    gender              = Gender(id=3),
+                    birth_date          = birth_date,
+                    recommender         = recommender,
+                    event_name          = event_name,
+                    grade               = Grade(id=1),
+                    terms_and_condition = TermsAndCondition(id=1),
+                    mileage             = 0
+                    )
```

Comment:
여주님! 회원가입할 때, gender, grade 데이터도 정규화해서 table을 만들어서 사용하신 것 너무 잘하셨습니다.
다만 전체 flow를 설명을 드리면, 
사용자가 처음에 회원가입 페이지에 들어오면, 프론트엔드에서 server에서 gender list와 grade list를 요청해서 받아가서 
drop down 버튼을 만들어줍니다.

그리고 회원가입을 진행하고 gender와 grade를 선택하고 회원가입 버튼을 누르면, 선택한 gender_id와 grade_id를 함께 서버에 보내줘야합니다. 
```suggestion
                User.objects.create(
                    account             = data['account'],
                    password            = hashed_pw,
                    name                = data['name'],
                    email               = data['email'],
                    phone_number        = data['phone_number'],
                    gender_id              = data['gender_id'],
                    birth_date          = birth_date,
                    recommender         = recommender,
                    event_name          = event_name,
                    grade_id               = data['grade_id']),
                    terms_and_condition = TermsAndCondition(id=1),
                    mileage             = 0
                    )
```

Reply:
병민님, 말씀하신 부분 숙지 후 수정 완료 했습니다. 

또한 아이디(account), 비밀번호 정규표현식 등에 잘못된 부분이 있어 추가로 수정했고,
회원가입 시에 주소지를 입력하는 부분이 있었는데 주소 테이블이 따로 빠져있어 누락이 되어 있었습니다.

주소 테이블이 유저 테이블을 참조하고 있는 관계이며, 데이터 입력 시 주소테이블로 데이터 연동될 수 있게끔 구현완료 했습니다.